File: Backend/RealTimeCharts.Application/Heart/Requests/GenerateHeartDataRequest.cs
using MediatR;
using OperationResult;
using RealTimeCharts.Shared.Enums;

namespace RealTimeCharts.Application.Heart.Requests
{
    public class GenerateHeartDataRequest : IRequest<Result>
    {
        public int Max { get; set; }
        public int Step { get; set; }
        public DataGenerationRate Rate { get; set; }
        public string ConnectionId { get; set; }
    }
}

File: Frontend/src/app/services/requests/generate-heart-data.request.ts
import { DataGenerationRate } from "src/app/models/data-generation-rate.enum";

export class GenerateHeartDataRequest {
    Max: number;
    Step: number;
    Rate: DataGenerationRate;
    ConnectionId: string;

    constructor(max: number, step: number, rate: DataGenerationRate, connectionId: string) {
        this.Max = max;
        this.Step = step;
        this.Rate = rate;
        this.ConnectionId = connectionId;
    }
}
